src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/RazorProjectService.cs

```diff
@@ -159,14 +159,25 @@ await _projectManager
     private void AddDocumentToMiscProjectCore(ProjectSnapshotManager.Updater updater, string filePath)
     {
         var textDocumentPath = FilePathNormalizer.Normalize(filePath);
+        _logger.LogDebug($"Asked to add {textDocumentPath} to the miscellaneous files project, because we don't have project info (yet?)");
 
-        _logger.LogDebug($"Adding {filePath} to the miscellaneous files project, because we don't have project info (yet?)");
-        var miscFilesProject = _projectManager.GetMiscellaneousProject();
+        var potentialProjects = _projectManager.FindPotentialProjects(textDocumentPath);
+        foreach (var project in potentialProjects)
+        {
+            if (project.DocumentFilePaths.Contains(textDocumentPath, FilePathComparer.Instance))
+            {
+                // Already in a known project, so we don't want it in the misc files project
+                _logger.LogDebug($"File {textDocumentPath} is already in {project.Key} so we're not adding it to the miscellaneous files project");
+                return;
+            }
+        }
```

Comment:
Isn't this basically the same as the below?

```suggestion
        if (_projectManager.TryResolveDocumentInAnyProject(textDocumentPath, _logger, out var _)
        {
        	// Already in a known project, so we don't want it in the misc files project
            _logger.LogDebug($"File {textDocumentPath} is already in {project.Key} so we're not adding it to the miscellaneous files project");
            return;
        }
```

Reply:
lol, it very well could be. That's a little embarassing  :D